code/modules/mining/smelter.dm

```diff
@@ -0,0 +1,265 @@
+/obj/machinery/smelter
+	name = "smelter"
+	desc = "An old Sendarian tool."
+	icon = 'icons/obj/blacksmithing_64.dmi'
+	icon_state = "forge"
+	density = TRUE
+	anchored = TRUE
+
+/obj/machinery/smelter/attackby(obj/item/weapon/W, mob/user, params)
+	var/smelting_result = W.on_smelt()
+	if(!smelting_result)
+		return ..()
+	if(user.drop_item())
+		to_chat(user, "You smelt [W].")
+		qdel(W)
+		var/obj/item/weapon/reagent_containers/glass/bucket/AB = new(get_turf(src))
+		AB.reagents.add_reagent(smelting_result, 75)
+		AB.reagents.chem_temp = 1000
+		AB.reagents.handle_reactions()
+		AB.name = "bucket of [AB.reagents.get_master_reagent_name()]"
+
+
+/obj/machinery/anvil
```

Comment:
Oh joy.

I've been wondering if any of the standard toolbox would get smithing molds though.

Also whether bluespace crystals are getting a smelting concept.